Update System.Linq.Async guidance to upgrade instead

### Type of issue

Outdated article

### Description

The recommended action here describes how you should use csproj exclusions to avoid System.Linq.Async exclusions.

Endjin released a new version of said package that will get rid of the conflicts in most cases automatically. Just need to upgrade it.

https://endjin.com/blog/2025/11/ix-v7-dotnet-10-linq-iasyncenumerable

> If you've been seeing errors of this kind since .NET 10 shipped:
>
> `error CS0121: The call is ambiguous between the following methods or properties: 'System.Linq.AsyncEnumerable.Select<TSource, TResult>(System.Collections.Generic.IAsyncEnumerable<TSource>, System.Func<TSource, TResult>)' and 'System.Linq.AsyncEnumerable.Select<TSource, TResult>(System.Collections.Generic.IAsyncEnumerable<TSource>, System.Func<TSource, TResult>)'`
>
> you may need to upgrade to v7 of Ix.NET's [System.Linq.Async](https://www.nuget.org/packages/System.Linq.Async/7.0.0) package. (In the long run you will want to stop using it entirely, and use the .NET runtime library [System.Linq.AsyncEnumerable](https://www.nuget.org/packages/System.Linq.AsyncEnumerable) package instead, but if you've ended up seeing these errors because of an indirect dependency, you might not be able to remove the reference just yet, in which case you'll need to upgrade it instead.)
>
> In most cases, that will solve the problem. It's possible you'll also need to add a reference to System.Interactive.Async v7 (or upgrade an existing reference to that version). There are also some more complex scenarios. This post explains what has changed and why.

### Page URL

https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/10.0/asyncenumerable

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/core/compatibility/core-libraries/10.0/asyncenumerable.md

### Document Version Independent Id

cabe10bd-a192-2e9f-df66-1d7b82fdc28d

### Platform Id

d72d0082-d90a-24e4-a73a-38c2d7696252

### Article author

@gewarren

### Metadata

* ID: 6bdd8cc4-3759-655d-bd5d-3599858b7a51
* PlatformId: d72d0082-d90a-24e4-a73a-38c2d7696252 
* Service: **dotnet-fundamentals**

[Related Issues](https://github.com/dotnet/docs/issues?q=is%3Aissue+is%3Aopen+cabe10bd-a192-2e9f-df66-1d7b82fdc28d)

---
[Associated WorkItem - 539265](https://dev.azure.com/msft-skilling/Content/_workitems/edit/539265)

--- a/docs/core/compatibility/core-libraries/10.0/asyncenumerable.md
+++ b/docs/core/compatibility/core-libraries/10.0/asyncenumerable.md
@@ -1,7 +1,7 @@
 ---
 title: "Breaking change - System.Linq.AsyncEnumerable in .NET 10"
 description: "Learn about the breaking change in .NET 10 where the AsyncEnumerable class is now included."
-ms.date: 2/21/2025
+ms.date: 12/11/2025
 ai-usage: ai-assisted
 ms.custom: https://github.com/dotnet/docs/issues/44886
 ---
@@ -32,7 +32,7 @@ This change can affect [source compatibility](../../categories.md#source-compati
 
 ## Recommended action
 
-If you're upgrading to .NET 10 and your code includes a direct package reference to `System.Linq.Async`, remove that package reference. For multitargeting both .NET 10 and a previous version, add a package reference to `System.Linq.AsyncEnumerable` instead.
+If you're upgrading to .NET 10 and your code includes a direct package reference to `System.Linq.Async`, remove that package reference or [upgrade to version 7.0.0](https://endjin.com/blog/2025/11/ix-v7-dotnet-10-linq-iasyncenumerable). For multitargeting both .NET 10 and a previous version, add a package reference to `System.Linq.AsyncEnumerable` instead.
 
 If `System.Linq.Async` is consumed indirectly via another package, avoid ambiguity errors by adding `<ExcludeAssets>` metadata with a value of `compile` or `all`:
 
@@ -56,12 +56,13 @@ If `System.Linq.Async` is consumed indirectly via another package, avoid ambigui
 
   Use this configuration only if you're certain no dependencies require System.Linq.Async at runtime.
 
-Most consuming code should be compatible without changes, but some call sites might need updates to refer to newer names and signatures. For example, a `Select` call like `e.Select(i => i * 2)` will work the same before and after. However, the call `e.SelectAwait(async (int i, CancellationToken ct) => i * 2)` will need to be changed to use `Select` instead of `SelectAwait`, as in `e.Select(async (int i, CancellationToken ct) => i * 2)`.
+Most consuming code should be compatible without changes, but some call sites might need updates to refer to newer names and signatures. For example, a `Select` call like `e.Select(i => i * 2)` works the same before and after. However, the call `e.SelectAwait(async (int i, CancellationToken ct) => i * 2)` needs to be changed to use `Select` instead of `SelectAwait`, as in `e.Select(async (int i, CancellationToken ct) => i * 2)`.
 
-Refer to the [System.Linq.AsyncEnumerable API documentation](xref:System.Linq.AsyncEnumerable) for the full set of LINQ extension methods available for <xref:System.Collections.Generic.IAsyncEnumerable`1>.
+For the full set of LINQ extension methods available for <xref:System.Collections.Generic.IAsyncEnumerable`1>, see the [System.Linq.AsyncEnumerable API documentation](xref:System.Linq.AsyncEnumerable).
 
 ## Affected APIs
 
 - <xref:System.Linq.AsyncEnumerable?displayProperty=fullName>
 - <xref:System.Collections.Generic.IAsyncEnumerable`1?displayProperty=fullName>
 - [System.Linq.Async package](https://www.nuget.org/packages/System.Linq.Async) (community-maintained)
+- [Ix.NET v7.0: .NET 10 and LINQ for IAsyncEnumerable\<T>](https://endjin.com/blog/2025/11/ix-v7-dotnet-10-linq-iasyncenumerable)

